File: 1869. Longer Contiguous Segments of Ones than Zeros/Solution.java
class Solution {
    public boolean checkZeroOnes(String s) {
        int[] res = new int[2];
        int i = 0;
        for (int j = 1; j <= s.length(); j++) {
            if (j == s.length() || s.charAt(j) != s.charAt(i)) {
                if (s.charAt(i) == '0') {
                    res[0] = Math.max(res[0], j - i);
                } else {
                    res[1] = Math.max(res[1], j - i);
                }
                i = j;
            }
        }
        return res[1] > res[0];
    }
}

File: 1869. Longer Contiguous Segments of Ones than Zeros/Solution.c
#include <math.h>
#include <stdbool.h>

bool checkZeroOnes(char *s)
{
    int res[2] = {0}, i, j;
    for (i = 0, j = 1; s[j]; j++) {
        if (s[j] == s[i])
            continue;
        if (s[i] == '0')
            res[0] = fmax(res[0], j - i);
        else
            res [1] = fmax(res[1], j - i);
        i = j;
    }
    if (s[i] == '0')
        res[0] = fmax(res[0], j - i);
    else
        res[1] = fmax(res[1], j - i);
    return res[1] > res[0];
}
